markdown.go

```diff
@@ -208,11 +231,52 @@ func (md *MarkdownFormatter) GenerateBodyBlock(args []*Argument) string {
 	if bodyArg != nil {
 		buf := new(bytes.Buffer)
 		fmt.Fprintf(buf, `
-#### Request Body
+### Request Body
 
-Argument "%s" is of file type. This endpoint expects a file in the body of the request as 'multipart/form-data'.
+Argument `+"`%s`"+` is of file type. This endpoint expects one or several files
+(depending on the command) in the body of the request as
+'multipart/form-data'.
 
 `, bodyArg.Name)
+
+		// Special documentation for /add
+		if bodyArg.Endpoint == "/api/v0/add" {
+			fmt.Fprintln(buf, html.EscapeString(`
+
+The `+"`add`"+` command not only allow adding files, but also uploading
+directories and complex hierarchies.
+
+This happens as follows: Every part in the multipart request is a *directory*
+or a *file* to be added to IPFS.
+
+Directory parts have a special content type `+"`application/x-directory`"+`.
+These parts do not carry any data. The part headers look as follows:
+
+`+"```"+`
+Content-Disposition: form-data; name="file"; filename="folderName"
+Content-Type: application/x-directory
+`+"```"+`
```

Comment:
note: these parts are also optional as directories will be auto-created based on the included files.

Reply:
Do I remember right that the `wrap` option is client-side only and users should manually add a top-level dir ?

Reply:
Wrap is server-side.